tests/test_pipeline_conversion.py

```diff
@@ -50,3 +50,18 @@ def test_pipeline_conversion_override_inputs(pipeline_overridden_config: Config)
     assert overridden["template"]["inputs"].get('training-images', [])[0] == 'overridden node image'
     assert len(overridden["template"]["inputs"].get('training-images', [])) == 1
     assert len(overridden["template"]["parameters"].items()) == 3
+
+
+def test_pipeline_parameter_conversion(pipeline_with_parameters_config):
+    for _name, pipe in pipeline_with_parameters_config.pipelines.items():
+        result = PipelineConverter(
+            config=pipeline_with_parameters_config,
+            commit_identifier="latest",
+        ).convert_pipeline(pipe)
+        assert isinstance(result["parameters"], dict)
+        parameter_name = "id"
+        assert result["parameters"][parameter_name]
+
+        parameter = result["parameters"][parameter_name]
+        assert parameter["config"]["targets"] or parameter["config"]["target"]
```

Comment:
At this point the target should have a name of `targets` and shouldn't have `target` in it at all. PipelineParameter parse method should take care of it.

We should also check that the targets is a list.

```suggestion
        assert parameter["config"]["targets"]
        assert not parameter["config"]["target"]
        assert isinstance(parameter["config"]["targets"], list)
```

Luckily, we have all these cases covered in `pipeline_with_parameters_config`

Reply:
Would be cool to have the test that target is removed:
        `assert not parameter["config"]["target"]`
I don't see this in the latest version.